### PR TITLE
08 restaurants page logic

### PR DESCRIPTION
**Pull request by AniSanikidze**
The data from json is displayed in restaurant cards.  Filtration of Prague College local restaurants by price range, vegan, vegetarian  and gluten fee options was implemented. The placeholder Image and logic for pagination were added(the pagination has some problems which should be fixed - #49).

### CODE REVIEW

**Review by Lucius1010011010 — DISMISSED** [review]
Restaurants are not visible potentially due to a proxy error; screenshots attached.

**Comment by AniSanikidze** [comment]
> Restaurants are not visible potentially due to a proxy error; screenshots attached.
> 
> 
> 

The restaurants are dependent on json provided from backend, so you need to run backend as well, otherwise, it won't work.

File: frontend/src/pages/Restaurants.js
Comment (adzai) [inline frontend/src/pages/Restaurants.js:60]: The logic is mostly fine, some small problems:
* vegan and vegetarian only worked when some price was checked -> there is a duplicate if statement
* trailing & -> for example prague-college/restaurants?vegan&
* vegan & vegetarian joined with a comma instead of & -> prague-college/restaurants/?vegan,vegetarian

I found out about all of these from backend logs

This is what fixed the issues above for me in Restaurants.js:
```javascript
  const featuredFilterValues = () => featured.map((item) => {
        return item
  })

  var chosenPriceRanges= priceRangeFilterValues();
  var chosenFeatured = featuredFilterValues();

 const showFilteredResults = () => {
    var pragueCollegeRestaurants = "/prague-college/restaurants?"
    if (chosenPriceRanges.length === 0 & chosenFeatured.length === 0) {
      return pragueCollegeRestaurants
    }
    if (chosenPriceRanges.length !== 0 & chosenFeatured.length !== 0) {
      return pragueCollegeRestaurants +=
            "price-range=" + chosenPriceRanges.join(",") + chosenFeatured.join("&")
    }
    if (chosenPriceRanges.length !== 0) {
      return pragueCollegeRestaurants +=
            "price-range=" + chosenPriceRanges.join(",")
    }
    if (chosenFeatured.length !== 0) {
        return pragueCollegeRestaurants + chosenFeatured.join("&")
   }
  }
```

In the future this still would have to be redone, because this is mostly hardcoded around price-range and there will be many more filters to apply.

Also, the page is called restaurants but displays only restaurants around Prague College. Wouldn't it be better to rename it, so this is more obvious to the user?

Comment (AniSanikidze) [reply frontend/src/pages/Restaurants.js:60]: Alright, I will fix those problems. Yes, you're right, the filtration is currently hardcoded. Because of the deadline of basic functionality, I focused on implementing only these 2 filters. I will try to make it more dynamic after we pass the basic functionality milestone.

I think it will be better to rename the header of the restaurants page rather than changing the page name in navbar.
Like this : 
image

Comment (adzai) [reply frontend/src/pages/Restaurants.js:60]: > I will try to make it more dynamic after we pass the basic functionality milestone.

Sounds good.

> I think it will be better to rename the header of the restaurants page rather than changing the page name in navbar.

Agreed, I think this is good for now.

**Review by Lucius1010011010 — CHANGES_REQUESTED** [review]
Please replace func main() in main.go with this:
func main() {
	args := os.Args[1:]
	if scraper.SliceContains(args, "--initialize") {
		conn, err := dbInitialise()
		if err != nil {
			log.Println(err)
			log.Fatal("Make sure the DB_DSN environment variable is set")
		} else {
			log.Println("Connection to postgres established, downloading data...")
		}
		err = storeRestaurants(conn)
		if err != nil {
			log.Fatal(err)
		}
	}
	r := mux.NewRouter()
	port := ":8080"
	r.HandleFunc("/prague-college/restaurants", pcRestaurantsHandler).Methods(http.MethodGet)
	r.HandleFunc("/restaurants", restaurantsHandler).Methods(http.MethodGet)
	r.PathPrefix("/").HandlerFunc(catchAllHandler)
	log.Println("Starting server on", port)
	log.Fatal(http.ListenAndServe(port, r))
}

It's from 07, and with that the server works properly. The logic itself works fine where it's implemented.

**Comment by AniSanikidze** [comment]
> Please replace func main() in main.go with this:
> func main() {
> args := os.Args[1:]
> if scraper.SliceContains(args, "--initialize") {
> conn, err := dbInitialise()
> if err != nil {
> log.Println(err)
> log.Fatal("Make sure the DB_DSN environment variable is set")
> } else {
> log.Println("Connection to postgres established, downloading data...")
> }
> err = storeRestaurants(conn)
> if err != nil {
> log.Fatal(err)
> }
> }
> r := mux.NewRouter()
> port := ":8080"
> r.HandleFunc("/prague-college/restaurants", pcRestaurantsHandler).Methods(http.MethodGet)
> r.HandleFunc("/restaurants", restaurantsHandler).Methods(http.MethodGet)
> r.PathPrefix("/").HandlerFunc(catchAllHandler)
> log.Println("Starting server on", port)
> log.Fatal(http.ListenAndServe(port, r))
> }
> 
> It's from 07, and with that the server works properly. The logic itself works fine where it's implemented.

I think it would be better to merge this code from main once http PR is merged.

**Review by Lucius1010011010 — APPROVED** [review]

**Review by AgiliaErnis — APPROVED** [review]

**Review by adzai — APPROVED** [review]
Great PR! Design looks good, logic works fine. 

The smaller things like pagination #49 and code improvements can be made
later.

**Pull request merged**
